autometa/common/external/diamond.py

```diff
@@ -29,146 +29,301 @@
+                "evalue": float(evalue),
+                "bitscore": float(bitscore),
+            }
+
+        Returns
+        -------
+        Dict
+            {qseqid: {DiamondResult}, ...}
+            
+        """
+        assert self == other_hit, f"qseqids do not match! {self} & {other_hit}"
         self.sseqids.update(other_hit.sseqids)
         return self
 
     def __sub__(self, other_hit):
-        assert self == other_hit, f'qseqids do not match! {self} & {other_hit}'
-        self.sseqids.pop(other_hit.sseqid)
+        """
+        Operator overloading to remove (subtract) the other sseqid hits dictionary from the sseqids dictionary
+        for a specific query sequence
+
+        Parameters
+        ----------
+        other_hit : Dict 
+            {
+            sseqid: {
+                "pident": float(pident),
+                "length": int(length),
+                "mismatch": int(mismatch),
+                "gapopen": int(gapopen),
+                "qstart": int(qstart),
+                "qend": int(qend),
+                "sstart": int(sstart),
+                "send": int(send),
+                "evalue": float(evalue),
+                "bitscore": float(bitscore),
+            }
+
+        Returns
+        -------
+        Dict
+            {qseqid: {DiamondResult}, ...}
+        """
+        assert self == other_hit, f"qseqids do not match! {self} & {other_hit}"
+        self.sseqids.pop(list(other_hit.sseqids.keys())[0])
         return self
 
     def get_top_hit(self):
-        top_bitscore = float('-Inf')
+        """
+        Returns the subject sequene ID with the highest bitscore amongst all the subject sequence that hit a query
+        """
+        top_bitscore = float("-Inf")
         top_hit = None
-        for sseqid,attrs in self.sseqids.items():
-            if attrs.get('bitscore', float('-Inf')) >= top_bitscore:
-                top_bitscore = attrs.get('bitscore')
+        for sseqid, attrs in self.sseqids.items():
+            if attrs.get("bitscore", float("-Inf")) >= top_bitscore:
+                top_bitscore = attrs.get("bitscore")
                 top_hit = sseqid
         return top_hit
 
-def makedatabase(fasta, database, nproc=1):
-    cmd = f'diamond makedb --in {fasta} --db {database} -p {nproc}'
-    logger.debug(f'{cmd}')
-    with open(os.devnull, 'w') as stdout, open(os.devnull, 'w') as stderr:
-        retcode = subprocess.call(cmd, stdout=stdout, stderr=stderr, shell=True)
-    if retcode:
-        raise OSError(f'DiamondFailed:\nArgs:{proc.args}\nReturnCode:{proc.returncode}')
+
+def makedatabase(fasta, database, nproc=mp.cpu_count()):
+    """
+    Creates a database against which the query sequence would be blasted
+
+    Parameters
+    ----------
+    fasta : str
+        Path to fasta file whose database needs to be made
+        e.g. '<path/to/fasta/file>'
+    database : str
+        Path to the output  nr database file
+        e.g. '<path/to/database/file>'
```

Comment:
These parameters are somewhat connected. You are referring to `nr` in the database parameter. But this function can handle _any_ fasta file and write out a respective diamond formatted database file from the provided `fasta`. I'm not sure `nr` should be mentioned here. The `database` parameter description should be more clear.